### PR TITLE
Donors now only ever see their own donation history

### LINKED ISSUE #5675: Donors should only ever see their own donation history
## User Story

As a donor], I want to only be shown my own donation history, regardless of DB errors so that I am not confused or able to glean private donation history information of other donors.

## Details

First spotted by @Benunc, this security issue appears to happen on corrupted DBs, which have failed to run all migrations successfully.

## Expected Behavior

Donors should only ever be able to see their own donation history.

## Steps to Reproduce

1. Import corrupted DB tables
2. Login as donor
3. View donor profile
4. Check donation history and dashboard giving stats for bugs

## Visuals

https://drive.google.com/file/d/13C7IQE3i4NYzA5qH9B__K387rk4qYE3R/view?usp=sharing

## Additional Context

Original Canny post: https://givewp.canny.io/admin/feedback/release-testing/p/donors-should-never-get-access-to-other-donors-information?boards=release-testing

## Acceptance Criteria

-   [x] Donors are only ever shown their own donation history
-   [x] If a DB error prevents donation history from being loaded correctly, an error message is displayed
-   [x] Donors are never able to view donation history of other donors

### PR DESCRIPTION
**Pull request by henryholtgeerts**
Resolves #5675 

## Description

This PR improves error handling within the Donations repository, so that a donor is only ever shown their own donation history.

## Affects

This PR affects some frontend rendering logic for the donation history tab, but primarily impacts logic in the Donations repository.

## Visuals

Screen Shot 2021-03-08 at 10 35 26 AM

## Testing Instructions

To easily test the frontend behavior without needing to corrupt a database, you can comment out if statement at line 53 in DonorProfiles/Tabs/DonationHistory/DonationsRoute.php. This will force an error response.

1) View the donor profile app
2) Check that an error appears in the dashboard
3) Check that an error appears in the Donation History tab

Uncomment the line (returning it to it's normal state).

1) View the donor profile app
2) Check that the dashboard works correctly
3) Check that the Donation History tab works correctly

## Pre-review Checklist

-   [x] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@since` tags included in DocBlocks
-   [x] Changes logged to the `Unreleased` section of `CHANGELOG.md`
-   [ ] Includes unit and/or end-to-end tests
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

### CODE REVIEW

**Comment by henryholtgeerts** [comment]
@Benunc I've opened this PR in draft mode, before I do some additional refactoring. If possible, I'd love to have you test this branch on the site where the donation history issue surfaced for you, to see if it persists after these changes. 

If the changes I've made resolve the problem, then I'll continue work on additionally cleaning up error handling logic in the Donor Dashboard domain before opening the PR for review.

cc: @JasonTheAdams

**Comment by Benunc** [comment]
This appears to leave the donor dashboard in a state of loading, never displaying anything. That's better than before, but still not ideal.

**Comment by henryholtgeerts** [comment]
@Benunc Thanks for testing it out! To clarify: does the dashboard become visible without any donations present, or are you just seeing the loading spinner and nothing appears beyond that?

**Comment by Benunc** [comment]
Screen Shot 2021-03-05 at 5 00 07 PM

@henryholtgeerts here's what it looks like:

**Comment by Benunc** [comment]
And @henryholtgeerts here's the console error: 

```
Uncaught (in promise) TypeError: Cannot read property 'message' of undefined
    at donor-profiles-app.js?ver=2.9.7:31
(anonymous) @ donor-profiles-app.js?ver=2.9.7:31
Promise.then (async)
Je @ donor-profiles-app.js?ver=2.9.7:31
(anonymous) @ donor-profiles-app.js?ver=2.9.7:31
(anonymous) @ donor-profiles-app.js?ver=2.9.7:31
935 @ donor-profiles-app.js?ver=2.9.7:31
t @ donor-profiles-app.js?ver=2.9.7:1
897 @ donor-profiles-app.js?ver=2.9.7:27
t @ donor-profiles-app.js?ver=2.9.7:1
(anonymous) @ donor-profiles-app.js?ver=2.9.7:1
(anonymous) @ donor-profiles-app.js?ver=2.9.7:1
```

File: src/DonorProfiles/Tabs/DonationHistoryTab/DonationsRoute.php [hunk 1]
```diff
@@ -43,11 +44,39 @@ public function handleRequest( $request ) {
...
+				]
+			);
+		}
+
+		return new WP_REST_Response(
+			[
+				'status'        => 400,
+				'response'      => 'database_error',
+				'body_response' => [
+					'message' => __( 'An error occured while retrieving your donation records.', 'give' ),
```
Comment (ravinderk) [inline src/DonorProfiles/Tabs/DonationHistoryTab/DonationsRoute.php:76]: I resolved typo in different branch https://github.com/impress-org/givewp/compare/e3d998a1e132%5E...19ac117ffc1f

**Review by ravinderk — APPROVED** [review]
@henryholtgeerts working fine for me.

**Review by Benunc — APPROVED** [review]
This loads up only the loading screen for corrupt data.

Screen Shot 2021-03-08 at 1 53 05 PM

I tested with several other donors on the site, and found similar results. 

This appears all set.

**Pull request merged**